File: 1999C/main.py
l=[]
t = int(input())
for i in range(t):
    n, s, m = map(int, input().split())
    segs = [(0, 0), (m, m)]

    for j in range(n):
        a, b = map(int, input().split())
        segs.append((a, b))

    segs.sort()

    for i in range(1, len(segs)):
        if segs[i][0] - segs[i - 1][1] >= s:
            print("YES")
            break
    else:
        print("NO")
# for i in l:
#     print(i)

File: 1999C/main.cpp
#include <iostream>
#include <vector>
#include <bits/stdc++.h>
using namespace std;
int main() {
    int t;
    cin >> t;
    while(t--){
        int n, s, m;
        cin >> n >> s >> m;
        vector<pair<int, int>> segs = {{0, 0}, {m, m}};

        for (int i = 0; i < n; ++i) {
            int a, b;
            cin >> a >> b;
            segs.push_back({a, b});
        }

        sort(segs.begin(), segs.end());


        bool canShower = false;
        for (int i = 1; i < segs.size(); ++i) {
            if (segs[i].first - segs[i - 1].second >= s) {
            cout << "YES" << endl;
            canShower = true;
            break;
            }
        }
        if (!canShower) {
            cout << "NO" << endl;
        }
        }

    // for(auto& i:v){
    //     cout << i << endl;
    // }
    
    return 0;
}



